src/components/PupilComponents/pupilUtils/lessonNavigation.tsx

```diff
@@ -0,0 +1,91 @@
+import { useRouter } from "next/router";
+import { useEffect, MouseEvent, useRef, useCallback } from "react";
+
+import { LessonSection, isLessonSection } from "../LessonEngineProvider";
+
+/**
+ * Subscribes to popstate events and calls the provided callback with the current section
+ */
+export function useLessonPopStateHandler(
```

Comment:
All of this would be unnecessary with Next 13's app router — but we're using the pages router which is outdated and geared toward static rendering.